test/utils/user_utils.test.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import assert from 'assert';
+
+import {Preferences} from 'constants';
+import {setLocalizeFunction} from 'utils/i18n_utils';
+import {displayUsername} from 'utils/user_utils';
+
+describe('user utils', () => {
+    const userObj = {
+        id: 100,
+        username: 'testUser',
+        nickname: 'nick',
+        first_name: 'test',
+        last_name: 'user',
+    };
+    it('should return username', () => {
+        assert.equal(displayUsername(userObj, 'UNKNOWN_PREFERENCE'), 'testUser');
+    });
+
+    it('should return nickname', () => {
+        assert.equal(displayUsername(userObj, Preferences.DISPLAY_PREFER_NICKNAME), 'nick');
+    });
+
+    it('should return fullname', () => {
+        assert.equal(displayUsername(userObj, Preferences.DISPLAY_PREFER_FULL_NAME), 'test user');
+    });
+
+    it('should return default username string', () => {
+        let noUserObj;
+        setLocalizeFunction(null);
```

Comment:
If I add a spy lib for the above test case then this can be removed

Reply:
Instead of making this test undo the changes made by the test case above, should we have an `afterEach` in the above test case that sets localize function back to `null`? (Isolating the impact of that test case's mocking?)